cmd/syncer/main.go

```diff
@@ -61,10 +67,14 @@ func main() {
 	ctx, log := logger.GetNewContextWithLogger()
 	log.Infof("Version : %s", syncer.Version)
 
+	// Set CO agnostic init params
 	clusterFlavor, err := config.GetClusterFlavor(ctx)
 	if err != nil {
 		log.Errorf("Failed retrieving cluster flavor. Error: %v", err)
 	}
+	common.SetInitParams(ctx, clusterFlavor, &syncer.COInitParams, *supervisorFSSName, *supervisorFSSNamespace,
```

Comment:
Looks like `SetInitParams` does not return anything and does not check `clusterFlavor` is empty or not.
Can you confirm what is the expected behavior when `err == nil` and `clusterFlavor` is nil in the previous step.
Shouldn't `SetInitParams` fail right away ?

Reply:
I added a fatal error message in `SetInitParams` if `clusterFlavor` is not recognized. As this is main function, I did not `exit` on failure.